option optimize_for = LITE_RUNTIME;


message HashedStorageHeader {
	enum HashAlgorithm {
		SHA1 = 1;
	}
	required HashAlgorithm hashAlgorithm = 1;
	required bytes hash = 2;
	required StorageHeader header = 3;
}

message StorageHeader {
    enum Operation {
        READ = 1;
        WRITE = 2;
        XOR_WRITE = 3;
    }
	required Operation operation = 1;
	required uint64 offset = 2;
	required uint64 length = 3;
	required uint64 requestTimestamp = 4;
}

message StorageResponseHeader {
	enum Status {
		OK = 1;
		ERROR = 2;
	}
	required Status status = 1;
	required StorageHeader header = 2;
}

message DictionaryResponseHeader {
    repeated HashedStorageHeader headers = 1;
    repeated string hosts = 3;
    repeated uint64 ports = 2;
}
